app/src/main/java/org/fossasia/pslab/fragment/SensorFragmentHMC5883L.java

```diff
@@ -0,0 +1,106 @@
+package org.fossasia.pslab.fragment;
+
+import android.os.AsyncTask;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v4.app.Fragment;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import org.fossasia.pslab.R;
+import org.fossasia.pslab.communication.ScienceLab;
+import org.fossasia.pslab.communication.peripherals.I2C;
+import org.fossasia.pslab.communication.sensors.HMC5883L;
+import org.fossasia.pslab.others.ScienceLabCommon;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+/**
+ * Created by asitava on 10/7/17.
+ */
+
+public class SensorFragmentHMC5883L extends Fragment {
+    private ScienceLab scienceLab;
+    private I2C i2c;
+    private SensorFragmentHMC5883L.SensorDataFetch sensorDataFetch;
+    private TextView tvSensorHMC5883Lbx;
+    private TextView tvSensorHMC5883Lby;
+    private TextView tvSensorHMC5883Lbz;
+
+    public static SensorFragmentHMC5883L newInstance() {
+        SensorFragmentHMC5883L sensorFragmentHMC5883L = new SensorFragmentHMC5883L();
+        return sensorFragmentHMC5883L;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        scienceLab = ScienceLabCommon.scienceLab;
+        i2c = scienceLab.i2c;
```

Comment:
well we would reach that state after scanning the sensors that implicitly means that device is connected. But there are cases in which its necessary.
Sorry for that, lets rectify the other one and we are good to merge this PR.